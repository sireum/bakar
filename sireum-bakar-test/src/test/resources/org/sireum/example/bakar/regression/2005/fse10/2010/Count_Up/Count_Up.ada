package Loops
is
  procedure Count_Up (N: in Integer; M: out Integer);
  --# derives M from N;
  --# pre N >= 0;
  --# post M = N;
end Loops;

package body Loops
is
  procedure Count_Up (N: in Integer; M: out Integer) is
    I : Integer;
  begin
    I := N;
    M := 0;
    while 0 < I loop
	    --# assert M = N - I
	    --#    and 0 < I and I <= N;
	    M := M + 1;
	    I := I - 1;
    end loop;
  end Count_Up;
end Loops;
