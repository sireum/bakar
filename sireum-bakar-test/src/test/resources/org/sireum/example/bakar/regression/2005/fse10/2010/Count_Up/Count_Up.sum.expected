-------------------------------------------------------------------------------
                          Semantic Analysis Summary                            
                                POGS GPL 2010                                  
            Copyright (C) 2010 Altran Praxis Limited, Bath, U.K.               
-------------------------------------------------------------------------------

Summary of:

Verification Condition files (.vcg)
Simplified Verification Condition files (.siv)
Proof Logs (.plg)
Dead Path Conjecture files (.dpc)
Summary Dead Path files (.sdp)

"status" column keys:
    1st character:
        '-' - No VC
        'S' - No SIV
        'U' - Undischarged
        'E' - Proved by Examiner
        'I' - Proved by Simplifier by Inference
        'X' - Proved by Simplifier by Contradiction
        'P' - Proved by Simplifier using User Defined Proof Rules
        'C' - Proved by Checker
        'R' - Proved by Review
        'F' - VC is False
    2nd character:
        '-' - No DPC
        'S' - No SDP
        'U' - Unchecked
        'D' - Dead path
        'L' - Live path

in the directory:
/Users/chalin/wkspdirs/36/sireum/sireum-spark-examples/src/org/sireum/profile/spark/examples/fse10/2010/Count_Up

Summary produced: 09-SEP-2010 16:52:23.61

File /Users/chalin/wkspdirs/36/sireum/sireum-spark-examples/src/org/sireum/profile/spark/examples/fse10/2010/Count_Up/loops/count_up.vcg
procedure Loops.Count_Up

VCs generated 09-SEP-2010 16:52:23

VCs simplified 09-SEP-2010 16:52:23

VCs for procedure_count_up :
 -----------------------------------------------------------------------------
| #   | From  | To                  | Proved By          | Dead Path | Status |
|-----------------------------------------------------------------------------
| 1   | start | rtc check @ 14      | Inference          | No DPC    |   I-   |
| 2   | start | rtc check @ 15      | Inference          | No DPC    |   I-   |
| 3   | start |    assert @ 17      | Inference          | No DPC    |   I-   |
| 4   | 17    |    assert @ 17      | Inference          | No DPC    |   I-   |
| 5   | 17    | rtc check @ 19      | Inference          | No DPC    |   I-   |
| 6   | 17    | rtc check @ 20      | Inference          | No DPC    |   I-   |
| 7   | start |    assert @ finish  | Inference          | No DPC    |   I-   |
| 8   | 17    |    assert @ finish  | Inference          | No DPC    |   I-   |
 -----------------------------------------------------------------------------


===============================================================================
Summary:

Proof strategies used by subprograms
-------------------------------------------------------------------------
Total subprograms with at least one VC proved by examiner:              0
Total subprograms with at least one VC proved by simplifier:            1
Total subprograms with at least one VC proved by contradiction:         0
Total subprograms with at least one VC proved with user proof rule:     0
Total subprograms with at least one VC proved using checker:            0
Total subprograms with at least one VC discharged by review:            0

Maximum extent of strategies used for fully proved subprograms:
-------------------------------------------------------------------------
Total subprograms with proof completed by examiner:                     0
Total subprograms with proof completed by simplifier:                   1
Total subprograms with proof completed with user defined rules:         0
Total subprograms with proof completed by checker:                      0
Total subprograms with VCs discharged by review:                        0

Overall subprogram summary:
-------------------------------------------------------------------------
Total subprograms fully proved:                                         1
Total subprograms with at least one undischarged VC:                    0
Total subprograms with at least one false VC:                           0
                                                                    -----
Total subprograms for which VCs have been generated:                    1


ZombieScope Summary:
-------------------------------------------------------------------------
Total subprograms for which DPCs have been generated:                   0
Total number subprograms with dead paths found:                         0
Total number of dead paths found:                                       0


VC summary:
-------------------------------------------------------------------------
Note: U/R denotes where the Simplifier has proved VCs using one or more user-
defined proof rules.

Total VCs by type:                                       
                            -----------Proved By Or Using------------
                     Total  Examiner Simp(U/R)  Checker Review False Undiscgd
Assert or Post:          4      0      4              0      0      0       0
Precondition check:      0      0      0              0      0      0       0
Check statement:         0      0      0              0      0      0       0
Runtime check:           4      0      4              0      0      0       0
Refinement VCs:          0      0      0              0      0      0       0
Inheritance VCs:         0      0      0              0      0      0       0
===============================================================================
Totals:                  8      0      8              0      0      0       0
% Totals:                       0%   100%             0%     0%     0%      0%
===================== End of Semantic Analysis Summary ========================
